File: sample-apps/JavalinPostgres/src/main/java/dev/aikido/handlers/SetUserHandler.java
package dev.aikido.handlers;

import dev.aikido.agent_api.SetUser;
import io.javalin.http.Context;
import io.javalin.http.Handler;

import static dev.aikido.agent_api.SetUser.setUser;

public class SetUserHandler implements Handler {
    @Override
    public void handle(Context ctx) throws Exception {
        String userId = ctx.header("user");
        if (userId != null) {
            // Useful for end-2-end tests:
            setUser(new SetUser.UserObject(userId, "John Doe"));
        }
    }
}

File: sample-apps/JavalinMySQLKotlin/src/main/kotlin/handlers/SetUserHandler.kt
package handlers

import dev.aikido.agent_api.SetUser
import dev.aikido.agent_api.SetUser.setUser
import io.javalin.http.Context
import io.javalin.http.Handler

class SetUserHandler : Handler {
    @Throws(Exception::class)
    override fun handle(ctx: Context) {
        val userId = ctx.header("user")
        if (userId != null) {
            // Useful for end-2-end tests:
            setUser(SetUser.UserObject(userId, "John Doe"))
        }
    }
}